### PR TITLE
Solved tasks with arrays

### PR DESCRIPTION
**Pull request by MelixoFF**

### CODE REVIEW

File: test/exercise/arrays/solution.rb [hunk 1]
```diff
@@ -2,11 +2,51 @@ module Exercise
   module Arrays
     class << self
       def replace(array)
-        array
+
+        def findMax(array)
```
Comment (greybutton) [inline test/exercise/arrays/solution.rb]: 1 объявлять функцию внутри функции не стоит
2 у руби другое соглашение об именовании, функций и переменных

File: test/exercise/arrays/solution.rb [hunk 2]
```diff
@@ -2,11 +2,51 @@ module Exercise
...
+          return maxValue
+        end
+
+        return array.map{|el| el>0 ? findMax(array) : el}
+        
       end
 
-      def search(_array, _query)
-        0
+      def search(array, query)
```
Comment (greybutton) [inline test/exercise/arrays/solution.rb]: попробуйте через рекурсию сделать

**Review by greybutton — CHANGES_REQUESTED** [review]

File: test/exercise/arrays/solution.rb (hunk 1 above)
Comment (MelixoFF) [reply test/exercise/arrays/solution.rb]: I know that declaring functions inside each other isn't a good idea, but for this test it's much easier to do like that

File: test/exercise/arrays/solution.rb (hunk 2 above)
Comment (MelixoFF) [reply test/exercise/arrays/solution.rb]: I don't like this realization with recursive function due to necessity of changing 2 parameters (left edge and right edge) but i've done it

File: test/exercise/arrays/solution.rb (hunk 1 above)
Comment (greybutton) [reply test/exercise/arrays/solution.rb]: what do you mean by that? how inside can be easier than outside? the place does not affect function declaration

File: test/exercise/arrays/solution.rb (hunk 2 above)
Comment (greybutton) [reply test/exercise/arrays/solution.rb]: okay, cool. but I think you kind of overthinking that solution. for example, you can set left_edge = 0 and right_edge = array.length - 1 by default. and I would say a perfect solution for the search is 4 lines of code. I don't ask you to do it in 4 lines, just some improvements if you don't mind

File: test/exercise/arrays/solution.rb [hunk 3]
```diff
@@ -1,12 +1,27 @@
 module Exercise
   module Arrays
     class << self
+
+      def find_max(array)
+        max_value = array[0]
+        for el in array
+          if el>max_value
```
Comment (aelaau) [inline test/exercise/arrays/solution.rb]: `el > max_value`

File: test/exercise/arrays/solution.rb [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
+            max_value=el
```
Comment (aelaau) [inline test/exercise/arrays/solution.rb]: max_value = el

File: test/exercise/arrays/solution.rb [hunk 5]
```diff
@@ -1,12 +1,27 @@
...
+          end
+        end
+        return max_value
+      end
+
       def replace(array)
-        array
+        return array.map{|el| el>0 ? find_max(array) : el}
```
Comment (aelaau) [inline test/exercise/arrays/solution.rb]: array.map { |el| el>0 ? find_max(array) : el }
Линтер не ругается на пробелы?

File: test/exercise/arrays/solution.rb [hunk 6]
```diff
@@ -1,12 +1,27 @@
...
       end
 
-      def search(_array, _query)
-        0
+      def search(array, query, left_edge=0)
+        return left_edge if array[0] == query
+        return -1 if array.length <= 1
+        centre = (0+(array.length-1))/2
+        return array[centre]>=query ? search(array[0,centre], query, left_edge)
```
Comment (aelaau) [inline test/exercise/arrays/solution.rb]: сделай через 
```
if pred
  return 1
else
  return 2
end
```
не стоит злоупотреблять тернарками, если строчки длинные получаются

**Review by aelaau — CHANGES_REQUESTED** [review]

File: test/exercise/arrays/solution.rb (hunk 5 above)
Comment (MelixoFF) [reply test/exercise/arrays/solution.rb]: No it doesn't

**Pull request closed without merging**